pkg/rancher-desktop/backend/backendHelper.ts

```diff
@@ -217,4 +228,122 @@ export default class BackendHelper {
 
     return availableVersions[0];
   }
+
+  /**
+   * Return a dictionary of all containerd shims installed in /usr/local/bin.
+   * Keys are the shim names and values are the filenames.
+   */
+  static async containerdShims(vmx: VMExecutor): Promise<Record<string, string>> {
+    const shims: Record<string, string> = {};
+
+    try {
+      const files = await vmx.execCommand({ capture: true }, '/bin/ls', '-1', '-p', '/usr/local/bin');
+
+      for (const file of files.split(/\n/)) {
+        const match = file.match(/^containerd-shim-([-a-z]+)-v\d+$/);
+
+        if (match) {
+          shims[match[1]] = file;
+        }
+      }
+    } catch (e: any) {
+      console.log('containerdShims: Got exception:', e);
+      throw e;
+    }
+
+    return shims;
+  }
+
+  /**
+   * Write a k3s manifest to define a runtime class for each installed containerd shim.
+   */
+  static async configureRuntimeClasses(vmx: VMExecutor) {
+    const runtimes = [];
+
+    for (const shim in await BackendHelper.containerdShims(vmx)) {
+      runtimes.push({
+        apiVersion: 'node.k8s.io/v1',
+        kind:       'RuntimeClass',
+        metaData:   {
+          name:    shim,
+          handler: shim,
+        },
+      });
+    }
+
+    if (runtimes.length > 0) {
```

Comment:
Is this condition backwards? We delete the file if it's not empty, and write (an always empty string) otherwise?

Reply:
Yes, that is wrong; thanks for catching it. I still have to do the Kubernetes BATS case for this.